### PR TITLE
Issue8

### PR DESCRIPTION
**Pull request by thomasing**

### CODE REVIEW

File: src/orbit.c [hunk 1]
```diff
@@ -18,6 +21,19 @@ orbit_t *orbit_create(const char *tle[])
 	struct orbit *m = (struct orbit*)malloc(sizeof(struct orbit));
 	if (m == NULL) return NULL;
 
+	m->time = NAN;
```
Comment (ryeng) [inline src/orbit.c]: NAN is float. C99 supports nan(""), which returns double.

File: src/orbit.c [hunk 2]
```diff
@@ -275,3 +256,32 @@ bool orbit_decayed(const orbit_t *orbit)
...
+	
+	double delta = ArcCos( vec3_dot(sol, earth) / vec3_length(sol) / vec3_length(earth) );
+	*depth = sd_earth - sd_sun - delta;
+
+	if (sd_earth < sd_sun) return false;
+	else if (*depth >= 0) return true;
+	else return false;
+}
+
+bool orbit_is_eclipsed(const orbit_t *orbit)
```
Comment (ryeng) [inline src/orbit.c:278]: Is this needed? Can't the caller just check orbit->eclipsed?

File: src/orbit.c [hunk 3]
```diff
@@ -275,3 +256,32 @@ bool orbit_decayed(const orbit_t *orbit)
...
+{
+	return orbit->eclipsed;
+}
+
+double orbit_eclipse_depth(const orbit_t *orbit)
```
Comment (ryeng) [inline src/orbit.c:283]: Is this needed? Can't the caller just check orbit->eclipse_depth?

File: src/orbit.c (hunk 2 above)
Comment (thomasing) [inline src/orbit.c:278]: See https://www.kernel.org/doc/Documentation/CodingStyle "Chapter 5: Typedefs". I would suggest that we keep orbit_t typedef and use accessor functions, or change to struct orbit and drop accessor functions. What do you think?

File: src/orbit.c (hunk 1 above)
Comment (thomasing) [inline src/orbit.c]: Fixed 3a81258

**Pull request merged**